### PR TITLE
Fix issues #2 and #4

### PR DESCRIPTION
**Pull request by andreiSaw**

### CODE REVIEW

**Comment by andreiSaw** [comment]
updated @sashabaranov

File: disneyland/server.go [hunk 1]
```diff
@@ -18,45 +18,42 @@ func detailedInternalError(err error) error {
 		fmt.Sprintf("Error processing job: %v", err),
 	)
 }
-func validateAccess(project string, userProject string, kind string, userKind string) (string, string, error) {
-	if kind == "" {
-		if userKind == "ANY" {
-			return "", "", grpc.Errorf(codes.DataLoss, "ListJobsKindRequest.Kind not specified")
-		}
-		kind = userKind
+func (s *Server) Init() {}
```
Comment (sashabaranov) [inline disneyland/server.go]: there should be empty lines before and after this method. Also Init() seem to be empty, maybe you should remove it.

File: disneyland/server.go [hunk 2]
```diff
@@ -18,45 +18,42 @@ func detailedInternalError(err error) error {
...
+func (u *User) ValidateUserAccess() bool {
```
Comment (sashabaranov) [inline disneyland/server.go]: This method is duplicated in user.go and server.go

File: disneyland/server.go [hunk 3]
```diff
@@ -18,45 +18,42 @@ func detailedInternalError(err error) error {
...
+	if u.ProjectAccess != "ANY" && u.KindAccess == "ANY" {
+		return true
 	}
-	if project == "" {
-		if userProject == "ANY" {
-			return "", "", grpc.Errorf(codes.DataLoss, "ListJobsKindRequest.Project not specified")
-		}
-		project = userProject
+	return false
+}
```
Comment (sashabaranov) [inline disneyland/server.go]: There should be one empty line between methods.

File: disneyland/server.go [hunk 4]
```diff
@@ -18,45 +18,42 @@ func detailedInternalError(err error) error {
...
-		}
-		project = userProject
-	}
-	if userProject != project && userProject != "ANY" {
-		return nil, grpc.Errorf(codes.PermissionDenied, "job.Project ≠ user.ProjectAccess")
-	}
-	in.Project = project
+
+	// if worker
+	if user.ValidateWorkerAccess() {
```
Comment (sashabaranov) [inline disneyland/server.go]: I guess `ValidateWorkerAccess` name does not make sense in this context.  `if user.IsWorker() {` might work much better

File: disneyland/server.go [hunk 5]
```diff
@@ -67,31 +64,31 @@ func (s *Server) CreateJob(ctx context.Context, in *Job) (*Job, error) {
...
 	job, err := s.Storage.GetJob(in.Id)
-	if userProject != job.Project && userProject != "ANY" {
-		err = grpc.Errorf(codes.PermissionDenied, "job.Project ≠ user.ProjectAccess")
-	}
+
 	if err != nil {
 		return nil, detailedInternalError(err)
 	}
 
+	if !user.ValidateJobAccess(job) {
```
Comment (sashabaranov) [inline disneyland/server.go]: Can look better `if !user.CanAccessJob(job) {`

File: disneyland/server.go [hunk 6]
```diff
@@ -120,21 +114,34 @@ func (s *Server) ModifyJob(ctx context.Context, in *Job) (*Job, error) {
...
+	pts, err := s.Storage.PullJobs(in.HowMany, in.Project, in.Kind)
 
-	project, kind, err := validateAccess(project, userProject, kind, userKind)
 	if err != nil {
 		return nil, detailedInternalError(err)
 	}
 
-	pts, err := s.Storage.PullJobs(in.HowMany, kind)
+	return &ListOfJobs{Jobs: pts}, nil
+}
```
Comment (sashabaranov) [inline disneyland/server.go]: spacing between methods

File: disneyland/storage.go [hunk 7]
```diff
@@ -219,5 +266,29 @@ func (storage *DisneylandStorage) PullJobs(how_many uint32, kind string) ([]*Job
 		return nil, err
 	}
 
-	return resultJobs, err
+	return ret, err
+}
+func (storage *DisneylandStorage) DeleteJob(id uint64, userProject string) (*Job, error) {
```
Comment (sashabaranov) [inline disneyland/storage.go]: spacing between methods

File: disneyland/storage.go [hunk 8]
```diff
@@ -172,45 +191,73 @@ func (storage *DisneylandStorage) UpdateJob(job *Job) (*Job, error) {
...
-			WHERE status=$1 and kind=$2
-			LIMIT $3
+			WHERE status=$1`
+	var rows *sql.Rows
+
+	if kind != "" {
+		strQuery += ` AND kind=$2`
+		if project != "" {
+			strQuery += ` AND project=$3`
+			strQuery = `	WITH updatedPts AS
```
Comment (sashabaranov) [inline disneyland/storage.go]: `=` after `+=`. Method became too monstrous, I recommend splitting it/removing nesting.

**Comment by andreiSaw** [comment]
@sashabaranov all fixed

**Pull request merged**